ros2param/ros2param/verb/set.py

```diff
@@ -56,23 +71,19 @@ def main(self, *, args):  # noqa: D102
             return 'Node not found'
 
         with DirectNode(args) as node:
-            parameter = Parameter()
-            Parameter.name = args.parameter_name
-            parameter.value = get_parameter_value(string_value=args.value)
-
+            parameters = self.build_parameters(args.parameters)
             response = call_set_parameters(
-                node=node, node_name=args.node_name, parameters=[parameter])
+                node=node, node_name=args.node_name, parameters=parameters)
+            results = response.results
 
-            # output response
-            assert len(response.results) == 1
-            result = response.results[0]
-            if result.successful:
-                msg = 'Set parameter successful'
-                if result.reason:
-                    msg += ': ' + result.reason
-                print(msg)
-            else:
-                msg = 'Setting parameter failed'
-                if result.reason:
-                    msg += ': ' + result.reason
-                print(msg, file=sys.stderr)
+            for result in results:
+                if result.successful:
+                    msg = 'Set parameter successful'
+                    if result.reason:
+                        msg += ': ' + result.reason
+                    print(msg)
+                else:
+                    msg = 'Setting parameter failed'
+                    if result.reason:
+                        msg += ': ' + result.reason
+                    print(msg, file=sys.stderr)
```

Comment:
These two blocks are very similar, with the exception of `successful` vs `failed` in the output string, and also using `stderr` or `stdout` for printing.  It feels like they can be combined, though I'm not sure of that so just take this as a suggestion.

Reply:
I've combined them [here](https://github.com/ros2/ros2cli/blob/2e900095aa87f9581c8dda0b9fa7d619034e6756/ros2param/ros2param/verb/set.py#L92-L88) but I feel like it's a little unnecessairly terse.

```python3
           for i, result in enumerate(results):
                print('Set parameter ' + parameters[i].name + ' ' +
                      'successful' if result.successful else 'failed' +
                      str(result.reason) if result.reason else '',
                      file=sys.stderr if result.successful else sys.stdout)
```

Reply:
Yeah, agreed.  This is kind of hard to read.  I'd say let's go back to the previous version (sorry).